UPGRADE-4.0.md

```diff
@@ -0,0 +1,207 @@
+# Upgrade from v3.x to v4.x
+
+Please consult the `examples` directory for reference example configurations. If you find a bug, please open an issue with supporting configuration to reproduce.
+
+## List of backwards incompatible changes
+
+- Minimum supported version of Terraform AWS provider updated to v4.6 to support the latest resources utilized
+- Minimum supported version of Terraform updated to v1.0
+- `ecs-instance-profile` sub-module has been removed; this functionality is available through the [`terraform-aws-modules/terraform-aws-autoscaling`](https://github.com/terraform-aws-modules/terraform-aws-autoscaling) module starting with version [v6.5.0](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/194). Please see the [`examples/ec2`](https://github.com/terraform-aws-modules/terraform-aws-ecs/tree/master/examples/ec2) example for a demonstration on how to use and integrate with the `terraform-aws-autoscaling` module.
```

Comment:
Maybe rename from `examples/ec2` to `examples/complete` to indicate that this is the main example (as we do in many other modules).